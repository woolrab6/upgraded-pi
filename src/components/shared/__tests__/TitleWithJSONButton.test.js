import React from 'react'
import {configure, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import {TitleWithJSONButton} from '../TitleWithJSONButton'

configure({adapter: new Adapter()})

it('renders button with given url', () => {
  const url = 'https://somebackend.xyz/ledger/12345'
  const title = 'Ledger'
  const btn = shallow(<TitleWithJSONButton title={title} url={url} />)
  expect(btn.find('JSONButton').props().url).toEqual(url)
  expect(btn.getElements()).toMatchSnapshot()
})
