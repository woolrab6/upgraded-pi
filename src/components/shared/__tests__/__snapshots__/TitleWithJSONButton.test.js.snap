// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with given url 1`] = `
Array [
  <div>
    <span>
      Ledger
    </span>
    <span
      className="pull-right"
    >
      <JSONButton
        url="https://somebackend.xyz/ledger/12345"
      />
    </span>
  </div>,
]
`;
